contracts/NFTPool.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.7.5;
+pragma experimental ABIEncoderV2;
+
+import {DaiPool, Erc20Pool, ReceiverWeight, Dai} from "./Pool.sol";
+import {IERC721} from "openzeppelin-contracts/token/ERC721/IERC721.sol";
+
+/// @notice NFT pool contract to support streaming based on NFT ownership
+/// A NFT can be a sender or a receiver, a unique id is generated based on
+/// NFT registry address and the tokenId
+contract NFTPool is Erc20Pool {
+    modifier nftOwner (address nftRegistry, uint tokenId) {
+        require(IERC721(nftRegistry).ownerOf(tokenId) == msg.sender, "not-NFT-owner"); _;
+    }
+    constructor(uint64 cycleSecs, Dai dai) Erc20Pool(cycleSecs, dai) {}
+
+    /// @notice generates a unique 20 bytes by hashing the nft registry  and tokenId
+    /// @param nftRegistry address of the NFT specific registry
+    /// @param tokenId the unique token id for the NFT registry
+    function nftID(address nftRegistry, uint128 tokenId) public pure returns (address id) {
+        // gas optimized without local variables
+        return address(uint160(uint256(
+                keccak256(abi.encodePacked(nftRegistry, tokenId)
+                ))));
+    }
+
+    /// @notice collect the funds of an NFT. Requires the msg.sender to own the NFT
+    /// @param nftRegistry address of the NFT specific registry
+    /// @param tokenId the unique token id for the NFT registry
+    function collect(address nftRegistry, uint128 tokenId) public nftOwner(nftRegistry, tokenId) {
+        uint128 collected = _collectInternal(nftID(nftRegistry, tokenId));
+        if (collected > 0) {
+            // msg.sender === nft owner
+            _transferToSender(msg.sender, collected);
+        }
+        emit Collected(msg.sender, collected);
```

Comment:
Ah, I think here the event should be the unique `nftID` and not the `msg.sender` who collected for that NFT